controllers/cinderapi_controller.go

```diff
@@ -256,18 +256,20 @@ func (r *CinderAPIReconciler) reconcileDelete(ctx context.Context, instance *cin
 
 			ks, err := keystone.GetKeystoneServiceWithName(ctx, helper, ksSvc["name"], instance.Namespace)
 
-			if err != nil {
-				if k8s_errors.IsNotFound(err) {
-					return ctrl.Result{}, nil
-				}
+			if err != nil && !k8s_errors.IsNotFound(err) {
 				return ctrl.Result{}, err
 			}
 
-			ksSvcObj := keystone.NewKeystoneService(ks.Spec, instance.Namespace, map[string]string{}, 10)
-			err = ksSvcObj.Delete(ctx, helper)
+			// Could get here if there was an error and it was "not found", so need to check again
+			// (we do it this way because we want to make sure the "RemoveFinalizer" call is
+			// executed even if the KeystoneService no longer exists)
+			if !k8s_errors.IsNotFound(err) {
```

Comment:
If `err` is `nil`, this returns `false`: https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsNotFound

Reply:
Thanks @abays, I tried this patch and it works as expected in a local environment, so +1 on my side.
I'll do the same in [1] and test.

[1] https://github.com/openstack-k8s-operators/glance-operator/pull/34

Reply:
Sounds good!